graph/pull.go

```diff
@@ -28,10 +18,24 @@ type ImagePullConfig struct {
 	OutStream   io.Writer
 }
 
+type Puller interface {
+	Pull(image, name, tag string, endpoint registry.APIEndpoint, imagePullConfig *ImagePullConfig, sf *streamformatter.StreamFormatter) (fallback bool, err error)
+}
+
+func (s *TagStore) NewPuller(endpoint registry.APIEndpoint, repoInfo *registry.RepositoryInfo) (Puller, error) {
```

Comment:
@tiborvass does the v2Puller or v1Puller require any private access to TagStore? If so maybe should enumerate those since it is the only thing keeping this logic in the graph package.